### PR TITLE
Add colors support to your io interface.

### PR DESCRIPTION
**Pull request by Song42**
* Color Control: SGR (Select Graphic Rendition)
* currently supports only 0-15 colors (format: ESC[;m)
* added atoi function for string manipulation
* added substring function for string manipulation
* separated print/println functions into another file

### CODE REVIEW

File: src/io/vga_buffer.rs
Comment (PfClaKr) [inline src/io/vga_buffer.rs:1]: just organised if else tree

**Review by PfClaKr — CHANGES_REQUESTED** [review]
i organised some if else tree, but not tested. thanks to before test this code !

```rust
	fn syntax_check(&mut self, s: &str) -> bool {
		use crate::include::string;
		if !s.contains('\x1b') || !s.contains('m') {
			return false;
		}
		if let Some(substr) = string::substring_between(s, '\x1b', 'm') {
			if substr.contains(' ') {
				return false;
			}
				let mut ss_chars = substr.chars();

				if ss_chars.next() != Some('['); {
					return false;
				}

				let color_code = ss_chars.next();
				if color_code.is_none() || !color_code.unwrap().is_ascii_digit() {
					return false;
				}

				let tbd = ss_chars.next();
				match tbd {
					Some(c) if c.is_ascii_digit() => {
						if ss_chars.next() == Some(';') {
							return ss_chars.next().is_none();
						}
					}
					Some(';') => {
						return ss_chars.next().is_none();
					}
					_ => return false,
				}
			}
			false
	}
				// what to do with this if/else forest?
				// let mut ss_chars = substr.chars();
				// if let Some(delim) = ss_chars.next() {
				// 	if delim == '[' {
				// 		if let Some(color_code) = ss_chars.next() {
				// 			if color_code.is_ascii_digit() {
				// 				if let Some(tbd) = ss_chars.next() {
				// 					if tbd.is_ascii_digit() {
				// 						if let Some(last) = ss_chars.next() {
				// 							if last == ';' {
				// 								let _end = ss_chars.next();
				// 								match _end {
				// 									Some(_end) => {
				// 										return false;
				// 									}
				// 									None => {
				// 										return true;
				// 									}
				// 								}
				// 							}
				// 						}
				// 					} else if tbd == ';' {
				// 						let _end = ss_chars.next();
				// 						match _end {
				// 							Some(_end) => {
				// 								return false;
				// 							}
				// 							None => {
				// 								return true;
				// 							}
				// 						}
				// 					}
				// 				}
				// 			}
				// 		}
				// 	}
				// }
			}
		}
		false
	}
```

**Comment by Song42** [comment]
Looks much better to read than before!

Thank you for the enhancement of my code!

**Review by Song42 — COMMENTED** [review]
Everything seems to be fine and looks good to me.
Thank You!

**Review by PfClaKr — APPROVED** [review]
Its good ! thanks a lot :)

**Pull request merged**